scripts/kata-arch.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+#
+# Copyright (c) 2017-2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -e
+
+typeset -r script_name=${0##*/}
+
+die()
+{
+    echo >&2 "ERROR: $*"
+    exit 1
+}
+
+usage()
+{
+cat <<EOT
+Description: Display host architecture name in various formats.
+
+Usage: $script_name [options]
+
+Options:
+
+ -d, --default         : Show arch(1) architecture (this is the default).
+ -g, --golang          : Show architecture name using golang naming.
+ -h, --help            : Show this help.
+ -k, --kernel          : Show architecture name compatible with Linux* build system.
+
+
+EOT
+}
+
+# Convert architecture to the name used by golang
+arch_to_golang()
+{
+	local -r arch="$1"
+
+	case "$arch" in
+		arm64|ppc64el) echo "$arch";;
+		x86_64) echo "amd64";;
+		*) die "unsupported architecture: $arch";;
+	esac
+}
+
+# Convert architecture to the name used by the Linux kernel build system
+arch_to_kernel()
+{
+	local -r arch="$1"
+
+	case "$arch" in
+		arm64|x86_64) echo "$arch";;
+		ppc64el) echo "powerpc";;
+		*) die "unsupported architecture: $arch";;
+	esac
+}
+
+main()
+{
+	local type="default"
+
+	local args=$(getopt \
+		-n "$script_name" \
+		-a \
+		--options="dghk" \
+		--longoptions="default golang help kernel" \
+		-- "$@")
+
+	eval set -- "$args"
+	[ $? -ne 0 ] && { usage; exit 1; }
```

Comment:
Yeah, I never quite know if I should use the `die()` here or not - I think using `die()` would actually just complicate things. I'll note that I think all the output in this failure case goes only to stdout though.

Reply:
Fixed.